integration-tests/client-rpc/test/hdwallet-management.test.ts

```diff
@@ -0,0 +1,197 @@
+import "mocha";
+import chaiAsPromised = require("chai-as-promised");
+import { use as chaiUse, expect } from "chai";
+import { RpcClient } from "./core/rpc-client";
+import { unbondAndWithdrawStake } from "./core/setup";
+import {
+	generateWalletName,
+	newWalletRequest,
+	newZeroFeeRpcClient,
+	shouldTest,
+	FEE_SCHEMA,
+	newWithFeeRpcClient,
+} from "./core/utils";
+chaiUse(chaiAsPromised);
+
+describe("Wallet management", () => {
+	let client: RpcClient;
+	before(async () => {
+		await unbondAndWithdrawStake();
+		if (shouldTest(FEE_SCHEMA.WITH_FEE)) {
+			client = newWithFeeRpcClient();
+		} else if (shouldTest(FEE_SCHEMA.ZERO_FEE)) {
+			client = newZeroFeeRpcClient();
+		}
+	});
+
+	it("cannot access un-existing wallet", async () => {
+		const nonExistingWalletName = generateWalletName();
+		const nonExistingWalletRequest = newWalletRequest(
+			nonExistingWalletName,
+			"123456",
+		);
+
+		await expect(
+			client.request("wallet_listStakingAddresses", [nonExistingWalletRequest]),
+		).to.eventually.rejectedWith(
+			`Invalid input: Wallet with name (${nonExistingWalletName}) not found`,
+		);
+		await expect(
+			client.request("wallet_listTransferAddresses", [nonExistingWalletRequest]),
+		).to.eventually.rejectedWith(
+			`Invalid input: Wallet with name (${nonExistingWalletName}) not found`,
+		);
+		await expect(
+			client.request("wallet_balance", [nonExistingWalletRequest]),
+		).to.eventually.rejectedWith(
+			`Invalid input: Wallet with name (${nonExistingWalletName}) not found`,
+		);
+		await expect(
+			client.request("wallet_transactions", [nonExistingWalletRequest]),
+		).to.eventually.rejectedWith(
+			`Invalid input: Wallet with name (${nonExistingWalletName}) not found`,
+		);
+	});
+
+	it("can create wallet with specified name", async () => {
+		const walletName = generateWalletName();
+		const walletRequest = newWalletRequest(walletName, "123456");
+
+		const walletCreateResult = await client.request("wallet_create", [
+			walletRequest
+			, "HD"
+		]);
+		let res= walletCreateResult.split(" ");
+		expect(res.length).to.deep.eq(24);
```

Comment:
why not `expect(walletCreateResponse).to.deep.eq(walletName);` ?

Reply:
Oh, there are 24 words returned?

Reply:
yes